lib/rucio/core/replica.py

```diff
@@ -748,12 +759,58 @@ def _list_replicas(dataset_clause, file_clause, state_clause, show_pfns, schemes
     for replicas in filter(None, files):
         for scope, name, bytes, md5, adler32, path, state, rse, rse_type, volatile in replicas:
 
+            pfns = []
+
             # reset the domain selection to original user's choice (as this could get overwritten each iteration)
             domain = deepcopy(original_domain)
 
-            pfns = []
-            if show_pfns and rse:
+            # if the file is a constituent, find the available archives and add them to the list of possible PFNs
+            if '%s:%s' % (scope, name) in constituents:
+                archive_result = list_replicas(dids=constituents['%s:%s' % (scope, name)],
+                                               schemes=schemes, client_location=client_location,
+                                               domain=domain, sign_urls=sign_urls,
+                                               signature_lifetime=signature_lifetime,
+                                               session=session)
+
+                # is it the only instance (i.e., no RSE for the replica)?
+                # then yield and continue as the replica does not exist and we don't want to return
+                # an additional empty pfn to the client
```

Comment:
It's not clear if this applies only to nested archive files, so it's better to clarify this comment: The pfn entry without replicas is there also to report the completness of a collection(dataset, container). It is compatible ?

Reply:
this is only in the archive workflow, and zips can never be datasets, only files, so it's ok.

Reply:
(and i still think that the empty pfn reply for collection completeness was a very bad design, we should get rid of that in the future)

Reply:
Which options would you propose then ? 
Resolving the content first and creating a dictionary with all entries (present or not) is intuitive no ? It does the job to report the completeness efficiently in one method without having to look-up the content separately. Feature requested by the wfms..

Reply:
to me, it breaks what list_replicas should do (= listing replicas) :-D

another call, is_complete(did) or similar would probably be the way go. i have to think about it.

Reply:
It gives also the details of which files is there or not for a collection at a site. Now if the collection concept is considered sub-optimal...